### PR TITLE
New 'code' authentication method

### PR DESCRIPTION
**Pull request by or-else**
The 'code' method is used to temporarily authenticate user for the purpose of changing a password. The secret code is sent to email/phone and stored in DB in k-v store under the key constructed from the email or phone.

### CODE REVIEW

File: server/auth/code/auth_code.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+	}
+
+	if ca.name != "" {
+		return errors.New("auth_code: already initialized as " + ca.name + "; " + name)
+	}
+
+	type configType struct {
+		// Length of the security code.
+		CodeLength int `json:"code_length"`
+		// Token expiration time.
```
Comment (aforge) [inline server/auth/code/auth_code.go:40]: Perhaps mention: "in seconds"

**Review by aforge — APPROVED** [review]
I assume we'll also need to make similar PCache changes to rethinkdb and mongodb adapters.

File: server/auth/code/auth_code.go (hunk 1 above)
Comment (or-else) [reply server/auth/code/auth_code.go]: done

File: server/auth/code/auth_code.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
+// Package code implements temporary no-login authentication by short numeric code.
+package token
```
[the remark this answers is not in the document]
Comment (or-else) [reply server/auth/code/auth_code.go]: done

File: server/store/store.go [hunk 3]
```diff
@@ -1052,6 +1053,43 @@ func (fileMapper) LinkAttachments(topic string, msgId types.Uid, attachments []s
 	return nil
 }
 
+// FilePersistenceInterface is an interface wchich defines methods used for file handling (records or uploaded files).
```
[the remark this answers is not in the document]
Comment (or-else) [reply server/store/store.go]: done

File: server/session.go [hunk 4]
```diff
@@ -850,26 +850,34 @@ func (s *Session) hello(msg *ClientComMessage) {
 
 // Account creation
 func (s *Session) acc(msg *ClientComMessage) {
+	newAcc := strings.HasPrefix(msg.Acc.User, "new")
+
 	// If token is provided, get the user ID from it.
```
[the remark this answers is not in the document]
Comment (or-else) [reply server/session.go]: done

File: server/db/mysql/schema.sql [hunk 5]
```diff
@@ -16,9 +16,11 @@ USE tinode;
 
 
 CREATE TABLE kvmeta(
-	`key` CHAR(32), 
+	`key` CHAR(64),
```
[the remark this answers is not in the document]
Comment (or-else) [reply server/db/mysql/schema.sql]: done

**Pull request merged**